### PR TITLE
feat: add record event support for redis destination

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## What are the changes introduced in this PR?

Add record event support for redis destination

## What is the related Linear task?

Resolves INT-978

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3124/test-report.html

File: src/v0/destinations/redis/transformV2.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+//     "fields": {
+//        MODEL_ID: '1691755780',
+//        VALID_AT: '2023-08-11T11:32:44.963062Z',
+//        USER_MAIN_ID: 'rid5530313526204a95efe71d98cd17d5a1',
+//        CHURN_SCORE_7_DAYS: 0.027986,
+//        PERCENTILE_CHURN_SCORE_7_DAYS: 0,
+//     },
+// }
+
+function handleRecordEventsForRedis(message, destination, metadata) {
```
Comment (saikumarrs) [inline src/v0/destinations/redis/transformV2.js:35]: There is no need to add "Redis" in the name. It's part of the Redis directory, so it's understood.

File: src/v0/destinations/redis/transform.js [hunk 2]
```diff
@@ -35,7 +36,10 @@ const transformSubEventTypeProfiles = (message, workspaceId, destinationId) => {
   // form the hash
   const hash = `${workspaceId}:${destinationId}:${message.context.sources.profiles_entity}:${message.context.sources.profiles_id_type}:${message.userId}`;
   const key = `${message.context.sources.profiles_model}`;
-  const value = JSON.stringify(message.traits);
+  let value = JSON.stringify(message.traits);
```
Comment (saikumarrs) [inline src/v0/destinations/redis/transform.js:39]: We should not JSON stringify traits by default if it is possible to not use them.

File: src/v0/destinations/redis/transformV2.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+const { transformSubEventTypeProfiles } = require('./transform');
```
Comment (saikumarrs) [inline src/v0/destinations/redis/transformV2.js:1]: Do we really need a "v2" file? It doesn't seem like a convention.

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: src/v0/destinations/redis/transformV2.js (hunk 3 above)
Comment (yashasvibajpai) [reply src/v0/destinations/redis/transformV2.js:1]: Fixed it to a utils.js file name

File: src/v0/destinations/redis/transform.js (hunk 2 above)
Comment (yashasvibajpai) [reply src/v0/destinations/redis/transform.js:39]: It was being stringified before also, should I remove the stringification?

Comment (saikumarrs) [reply src/v0/destinations/redis/transform.js:39]: The implementation should have looked something like this,

```
if event type is record {
  // do record specific logic
} else {
  // stringify traits
}
```

File: src/v0/destinations/redis/transformV2.js (hunk 1 above)
Comment (saikumarrs) [reply src/v0/destinations/redis/transformV2.js:35]: Maybe you haven't pushed the changes to upstream.

**Review by koladilip — COMMENTED** [review]
I am thinking of adding action to all cases and base the rudder-server logic on that

**Comment by yashasvibajpai** [comment]
> I am thinking of adding action to all cases and base the rudder-server logic on that 
@koladilip 
But if we mandate action in all events by default, in case of non-profiles standard identify events, older implementation of using HMSET to replace the whole hash would become redundant

File: src/v0/destinations/redis/transform.js (hunk 2 above)
Comment (yashasvibajpai) [reply src/v0/destinations/redis/transform.js:38]: Right now the flow is
event -
- if event is a record type event
- - handle insert/delete action related logic

- if event is having profiles' contextual data
- -  it is a profiles-identify event (backward compatible, only insertion supported)

- for none of the above
-- it is a standard redis destination's identify event

**Comment by koladilip** [comment]
> > I am thinking of adding action to all cases and base the rudder-server logic on that
> > @koladilip
> > But if we mandate action in all events by default, in case of non-profiles standard identify events, older implementation of using HMSET to replace the whole hash would become redundant

Lets discuss over all with @saikumarrs

File: src/v0/destinations/redis/transform.js (hunk 2 above)
Comment (saikumarrs) [reply src/v0/destinations/redis/transform.js:38]: I don't think you're getting it.

I meant that you should not stringify traits by default. Instead, do it only for non-record type of event flows.

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Pull request closed without merging**